### PR TITLE
Updated method of modelchain.ModelChain.prepare_inputs in modelchain.py

### LINKED ISSUE #523: get_solarpostion in modelchain.prepare_inputs does not use weather data.
**Problem**
``get_solarpostion`` method in ``modelchain.ModelChain.prepare_inputs`` does not use weather data.

**To Reproduce**
change line 745 - 746
``` modelchain.py
755:       self.solar_position = self.location.get_solarposition(
756:            self.times, method=self.solar_position_method)
```

to 

``` modelchain.py
755:       self.solar_position = self.location.get_solarposition(
756:            self.times, method=self.solar_position_method,
757:            pressure=self.weather.pressure,
758:            temperature=self.weather.temp_air)
```

**Versions:**
 - ``pvlib.__version__``:  0.5.2
 - ``pandas.__version__``: 0.23.3
 - python: 3.5.2

### PR DESCRIPTION
**Pull request by Ahanmr**
#523
Added weather data arguments in get_solarpostion method of modelchain.ModelChain.prepare_inputs as part of the Pull request.

 - [x] Closes #523 
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

### CODE REVIEW

**Comment by Ahanmr** [comment]
This issue closes #523

**Comment by cwhanse** [comment]
@Ahanmr thanks for the contribution. Looking at the test failures, the code needs to handle the case where `pressure` or `temp_air` is not in `weather`. @wholmgren suggested using `pvlib.tools._build_kwargs` here, which I think would work nicely. There's an example [here](https://github.com/pvlib/pvlib-python/blob/245ace1a2eb5c03ec355566a80635c9ed14b2441/pvlib/pvsystem.py#L402). 

In this context, the code would be like:

```
    kwargs = _build_kwargs(['pressure', 'temp_air'], weather)
    kwargs['temperature'] = kwargs.pop('temp_air')
    self.solar_position = self.location.get_solarposition(self.times,
        method=self.solar_position_method, **kwargs)
```

@wholmgren note the unfortunate need to rename.

**Comment by Ahanmr** [comment]
@cwhanse I've made the necessary changes. Thank you for letting me know about the pvlib.tools._build_kwargs which help me pass keyword arguments in case some are missing. But I still feel I missed something, I'll try fixing it meanwhile.

File: pvlib/modelchain.py [hunk 1]
```diff
@@ -860,8 +861,12 @@ def prepare_inputs(self, weather, times=None):
 
         self.times = self.weather.index
 
+        kwargs = _build_kwargs(['pressure', 'temp_air'], weather)
+        kwargs['temperature'] = kwargs.pop('temp_air')
```
Comment (wholmgren) [inline pvlib/modelchain.py]: this raises a `KeyError` when `temp_air` does not exist. Use `try: ... except KeyError: pass`

**Pull request closed without merging**